packages/@expo/server/src/index.ts

```diff
@@ -152,6 +147,42 @@ export function createRequestHandler(
 
     debug('Request', sanitizedPathname);
 
+    if (routesManifest.rewrites) {
+      for (const route of routesManifest.rewrites) {
+        if (!route.namedRegex.test(sanitizedPathname)) {
+          continue;
+        }
+
+        const url = getRedirectRewriteLocation(request, route);
+
+        if (url) {
+          request = new Request(new URL(url, new URL(request.url).origin), request);
+        }
+      }
+    }
+
+    if (routesManifest.redirects) {
+      for (const route of routesManifest.redirects) {
+        if (!route.namedRegex.test(sanitizedPathname)) {
+          continue;
+        }
+
+        const Location = getRedirectRewriteLocation(request, route);
+
+        if (Location) {
+          debug('Redirecting', Location);
+
+          // Get the params
+          return new Response(null, {
+            status: route.permanent ? 308 : 307,
```

Comment:
Should we alter this depending on the method? I think it might be nicer to have special cases for `GET` to use the lower codes, potentially